blog.qmd

```diff
@@ -0,0 +1,228 @@
+           docid_field = 'course_id') %>% 
+  janitor::clean_names() %>% 
+  filter(course_status == "ACTIVE",
+         type_of_credit == "COLLEGE CREDIT")
+```
+
+```{r}
+#| echo: false
+courses %>% 
+  slice(1:1000) %>% 
+  gt() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) %>% 
+  my_theme()
+```
+
+The first step in our analysis is to create a corpus, a collection of text documents, from our course descriptions. We can achieve this using the corpus() function in quanteda. Next, we extract the tokens in the corpus—usually words, but they can also be n-grams or multi-word expressions. The tokens function allows us to define what we mean by tokens and apply some rules to ignore elements such as punctuation and digits.
+
+```{r}
+corpus <- corpus(courses)
+```
+
+```{r}
+#| echo: false
+corpus[1:5]
+```
+
+```{r}
+tk <- tokens(corpus, what = "word", remove_punct = TRUE, remove_numbers = TRUE)
+```
+
+```{r}
+#| echo: false
+tk[1:5]
+```
+
+## Key-Term Searches with a Dictionary
+
+Our primary interest lies in identifying courses related to different types of work-based learning such as internships, apprenticeships, or practicums. For each type of work-based learning experience, we create a list of terms that we want to treat equivalently. For instance, our dictionary can specify that a course description refers to a clinical WBL if either of the terms "clinicals" or "clinical experience" appear.
+
+```{r}
+dict <-
+  dictionary(list(apprenticeship = "apprentice*",
+                  practicum = c("practicum", "practica"),
+                  coop = c("co-op", "cooperative education", "co-operative education"),
+                  clinicals = c('clinicals', 'clinical experience'),
+                  on_the_job = c("on the job training", "job training", "on-the-job training"),
+                  wbl = c('work-based learning', "work based learning", "wbl"),
+                  real_world_experience = c("real-world experience", "real world experience"),
+                  service_learning = "serive learning",
+                  field_experience = c("fieldwork", "field experience", "field-experience")))
+```
+
+Armed with our dictionary, we're ready to search for the terms using the kwic() (key-word in context) function. This function takes in a corpus and a dictionary as inputs, along with a window parameter specifying the number of tokens before and after a keyword that we want to see for context.
+
+```{r}
+keywords <- kwic(corpus, dict, window = 10) %>% as_tibble()
+```
+
+```{r}
+#| echo: false
+keywords %>% 
+  gt() %>% 
+  my_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+Finally, we can join the results from the dictionary-based keyword in-context search back to the course level data and perform some wrangling to analyze the prevalence of work-based learning opportunities in Florida's community colleges.
+
```

Comment:
can we be more specific than "some wrangling"? E.g. reshaping, appending, whatever more specific programming verbs you used here :)

Reply:
I actually think this is too much detail now given how much we added in other areas. The last sentence probably isn't needed, and I think the previous ones that come after "data transformations" can probably be collapsed into one sentence that just highlights the data transformation verbs ("split", "create new columns", that kind of thing). Does that make sense?

Reply:
Simplified the paragraph!